### PR TITLE
Tests for .NET aliases

### PR DESCRIPTION
**Pull request by mikhailshilkov**
Adding tests for aliases in .NET - both tests fail. Aliases not working yet? @CyrusNajmabadi

### CODE REVIEW

File: tests/integration/aliases/dotnet/adopt_into_component/step2/Program.cs [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+{
+    public Component4(string name, ComponentResourceOptions options = null) 
+        : base("my:module:Component4", name, 
+            ComponentResourceOptions.Merge(
+                new ComponentResourceOptions
+                {
+                    Aliases =
+                    {
+                        new Alias { Parent = null /* What's pulumi.rootStackResource in .NET? */ },
+                        new Alias { Parent = null /* What's pulumi.rootStackResource in .NET? */ }
```
Comment (CyrusNajmabadi) [inline tests/integration/aliases/dotnet/adopt_into_component/step2/Program.cs]: use `NoParent = true` :)

Comment (mikhailshilkov) [reply tests/integration/aliases/dotnet/adopt_into_component/step2/Program.cs]: Thanks, changed. Aliases still fail all over the place.

**Comment by CyrusNajmabadi** [comment]
I'll look into what's going wrong here.  thanks for writing the tests!

File: tests/integration/aliases/dotnet/adopt_into_component/step2/Program.cs [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+class Component3 : ComponentResource
+{
+    public Component3(string name, ComponentResourceOptions options = null) 
+        : base("my:module:Component3", name, options)
+    {
+        new Component2(name + "-child",
+            new ComponentResourceOptions 
+            { 
+                Aliases = { new Alias { Parent = options?.Parent } },
```
Comment (CyrusNajmabadi) [inline tests/integration/aliases/dotnet/adopt_into_component/step2/Program.cs]: this likely won't work.  if `options?.Parent` evaluates to null this effectively makes a do-nothing alias.  you'd have to write this as:

```new Alias { Parent = options?.Parent, NoParent = options?.Parent == null }```

**Comment by CyrusNajmabadi** [comment]
Ok.  investigating why this is failing.

File: sdk/dotnet/Pulumi/Core/Urn.cs [hunk 3]
```diff
@@ -8,17 +8,17 @@ namespace Pulumi
...
-    internal static class Urn
+    public static class Urn
     {
         /// <summary>
         /// Computes a URN from the combination of a resource name, resource type, optional parent,
         /// optional project and optional stack.
         /// </summary>
         /// <returns></returns>
-        internal static Output<string> Create(
+        public static Output<string> Create(
```
Comment (mikhailshilkov) [inline sdk/dotnet/Pulumi/Core/Urn.cs]: Exposed this method to use it in one of the tests

**Comment by mikhailshilkov** [comment]
@CyrusNajmabadi I fixed the existing and included the missing tests. Review for review! Well done fixing it!

File: sdk/dotnet/Pulumi/Core/Urn.cs (hunk 3 above)
Comment (pgavlin) [reply sdk/dotnet/Pulumi/Core/Urn.cs]: Any other way to consume this from the test without exposing the method?

Comment (mikhailshilkov) [reply sdk/dotnet/Pulumi/Core/Urn.cs]: Our integration tests are just Pulumi programs, so I don't think so... FWIW, the same method is exposed in the Node.js SDK too.

**Pull request merged**